src/librustc/ty/print/pretty.rs

```diff
@@ -1527,12 +1530,54 @@ define_print_and_forward_display! {
         p!(print_def_path(self.def_id, self.substs));
     }
 
-    &'tcx ty::Const<'tcx> {
-        match self.val {
-            ConstValue::Unevaluated(..) |
-            ConstValue::Infer(..) => p!(write("_")),
-            ConstValue::Param(ParamConst { name, .. }) => p!(write("{}", name)),
-            _ => p!(write("{:?}", self)),
+    ty::Const<'tcx> {
+        match (self.val, &self.ty.sty) {
+            | (ConstValue::Unevaluated(..), _)
+            | (ConstValue::Infer(..), _)
+            => p!(write("_: "), print(self.ty)),
```

Comment:
`Unevaluated` should print the path if it's not `AnonConst`. (and if it is, maybe it could pretty-print the HIR or something - or at least do the closure thing, with a span)

Reply:
Do you want to maybe leave a comment about this?